# 회귀분석 8장 과제

> "작성 완료"

- toc: true
- branch: master
- badges: true
- comments: true
- author: 한재수
- categories: [R, statistics]

## 8.1

In [29]:
# 8.1
x1 <- c(175, 210, 192, 250, 245, 226, 260, 230, 205, 185)
x2 <- c(15, 18, 20, 28, 25, 21, 32, 25, 24, 19)
y <- c(120.4, 112.5, 95.4, 162.3, 160.2, 131.5, 157.6, 158.4, 149.6, 130.4)
X <- t(rbind(rep(1, length(x1)), x1, x2))
m0 <- lm(y~1)

In [31]:
# 1)
m1 <- lm(y~x1+x2)
summary(m1)


Call:
lm(formula = y ~ x1 + x2)

Residuals:
    Min      1Q  Median      3Q     Max 
-30.261  -9.727   6.541  10.963  12.206 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)
(Intercept)  35.3359    48.5371   0.728    0.490
x1            0.2048     0.4085   0.501    0.632
x2            2.5506     2.3334   1.093    0.311

Residual standard error: 16.18 on 7 degrees of freedom
Multiple R-squared:  0.6255,	Adjusted R-squared:  0.5185 
F-statistic: 5.845 on 2 and 7 DF,  p-value: 0.03215


`-` $\hat{y}=33.3359 + 0.2048x_1 + 2.5506x_2$

In [32]:
# 2)
Var_b = solve(t(X)%*%X)
Var_b0 = Var_b[1,1]; Var_b0
Var_b1 = Var_b[2,2]; Var_b1
Var_b2 = Var_b[3,3]; Var_b2

[1] 8.999748

[1] 0.000637595

[1] 0.02079972

$Var(b_0) = 8.99974754425673\sigma^2\\
Var(b_1)=0.000637594973081464\sigma^2\\
Var(b_2)=0.0207997160628333\sigma^2$

`3)`

`-` b1 : 시간을 고정시킨 상태에서 온도가 1도 상승했을때 인장강도의 변화

`-` b2 : 공정온도를 고정시킨 상태에서 시간이 1분 더 지났을때 인장강도의 변화

In [34]:
# 4)
confint(m1, level = 0.95)

2.5 %       97.5 %    
(Intercept) -79.4362406 150.107949
x1           -0.7612855   1.170788
x2           -2.9669628   8.068228

`-` $b_1$의 95% 신뢰구간은 (-0.7612855, 1.170788)

`-` $b_2$의 95% 신뢰구간은 (-2.9669628, 8.068228)

In [38]:
# 5)
anova(m0, m1)

Res.Df RSS      Df Sum of Sq F       Pr(>F)    
1 9      4892.701 NA       NA       NA         NA
2 7      1832.382  2 3060.319  5.84546 0.03214666

`-` p-value가 0.05보다 작으므로 영가설을 기각한다, 회귀모형은 유의하다

In [39]:
#6)
predict(m1, data.frame(x1 = 200, x2 = 20), level = 0.95, interval = 'confidence')

fit      lwr      upr     
1 127.2987 112.8839 141.7135

`-` $x_1=200,\;x_2=20\text{에서 }E(y)$의 95% 신뢰구간은 (112.8839, 141.7135)이다

In [50]:
#7)
q = cbind(c(0, 1, -1))
b = cbind(c(33.3359, 0.2048, 2.5506))
t = (t(q) %*% b) / sqrt(t(q) %*% solve(t(X) %*% X) %*% q * mse)
abs(t) 
qt(0.975, 7) 

[,1]    
[1,] 0.867669

[1] 2.364624

`-` 검정통계량 t값이 2.36보다 작으므로 영가설을 기각하지 못한다

`-` $\beta_1, \beta_2$는 같지 않다고 할 수 없다

In [51]:
#8)
predict(m1, data.frame(x1 = 240, x2 = 30), level = 0.95, interval = 'predict')

fit      lwr      upr     
1 160.9951 114.5839 207.4063

$x_1=240,\;x_2=30\text{에서 }y$의 95% 예측구간은 (114.5839, 207.4063)이다

## 8.10

In [52]:
# 8.10

x1 <- c(2, 2, 3, 3, 2, 2, 3, 3)
x2 <- c(4, 4, 4, 4, 6, 6, 6, 6)
y <- c(39, 42, 51, 48, 53, 49, 61, 60)
X <- t(rbind(rep(1, length(x1)), x1, x2))
m0 <- lm(y~1)

In [53]:
# 1)
m1 <- lm(y~x1+x2); summary(m1)


Call:
lm(formula = y ~ x1 + x2)

Residuals:
     1      2      3      4      5      6      7      8 
-1.375  1.625  1.375 -1.625  1.875 -2.125  0.625 -0.375 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept)   0.3750     4.7405   0.079 0.940016    
x1            9.2500     1.3276   6.968 0.000937 ***
x2            5.3750     0.6638   8.097 0.000466 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.877 on 5 degrees of freedom
Multiple R-squared:  0.958,	Adjusted R-squared:  0.9412 
F-statistic: 57.06 on 2 and 5 DF,  p-value: 0.000361


`-` $\hat{y}=0.375 + 0.925x_1 + 5.375x_2$

In [55]:
# 2)
m2 <- lm(y~x1); summary(m2)


Call:
lm(formula = y ~ x1)

Residuals:
   Min     1Q Median     3Q    Max 
-7.000 -4.688 -0.250  5.250  7.250 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)  
(Intercept)   27.250     11.608   2.348   0.0572 .
x1             9.250      4.553   2.032   0.0885 .
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 6.439 on 6 degrees of freedom
Multiple R-squared:  0.4076,	Adjusted R-squared:  0.3088 
F-statistic: 4.128 on 1 and 6 DF,  p-value: 0.08846


`-` $\hat{y}=27.25 + 9.25x_1$

In [56]:
anova(m2)

Df Sum Sq  Mean Sq   F value  Pr(>F)    
x1        1  171.125 171.12500 4.127638 0.08846031
Residuals 6  248.750  41.45833       NA         NA

`-` p-value가 0.05보다 크다 

`-` $x_1$은 회귀모형에서 유의하지 않은 변수이다

In [58]:
# 3)
m3 <- lm(y~x2); summary(m3)


Call:
lm(formula = y ~ x2)

Residuals:
   Min     1Q Median     3Q    Max 
-6.750 -3.750  0.125  4.500  6.000 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)  
(Intercept)   23.500     10.111   2.324   0.0591 .
x2             5.375      1.983   2.711   0.0351 *
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 5.609 on 6 degrees of freedom
Multiple R-squared:  0.5505,	Adjusted R-squared:  0.4755 
F-statistic: 7.347 on 1 and 6 DF,  p-value: 0.03508


`-` $\hat{y}=23.5 + 5.375x_2$

In [59]:
anova(m3)

Df Sum Sq  Mean Sq   F value Pr(>F)    
x2        1  231.125 231.12500 7.34702 0.03508095
Residuals 6  188.750  31.45833      NA         NA

`-` p-value가 0.05보다 작다 

`-` $x_2$은 회귀모형에서 유의한 변수이다

In [62]:
# 4)
m_beta <- lm(scale(y)~scale(x1)+scale(x2)); summary(m_beta) # 표준화된 회귀모형


Call:
lm(formula = scale(y) ~ scale(x1) + scale(x2))

Residuals:
       1        2        3        4        5        6        7        8 
-0.17754  0.20982  0.17754 -0.20982  0.24210 -0.27438  0.08070 -0.04842 
attr(,"scaled:center")
[1] 50.38
attr(,"scaled:scale")
[1] 7.745

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 9.329e-17  8.571e-02   0.000 1.000000    
scale(x1)   6.384e-01  9.163e-02   6.968 0.000937 ***
scale(x2)   7.419e-01  9.163e-02   8.097 0.000466 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.2424 on 5 degrees of freedom
Multiple R-squared:  0.958,	Adjusted R-squared:  0.9412 
F-statistic: 57.06 on 2 and 5 DF,  p-value: 0.000361


`-` $\hat{y}=0.6384w_1 + 0.7419w_2$

In [64]:
# 5)
anova(m_beta)

Df Sum Sq    Mean Sq    F value  Pr(>F)      
scale(x1) 1  2.8529324 2.85293242 48.54610 0.0009365829
scale(x2) 1  3.8532301 3.85323013 65.56738 0.0004657066
Residuals 5  0.2938375 0.05876749       NA           NA

`-` p-value가 매우 작다

`-` 회귀모형은 유의하다

In [65]:
# 6)
cor(X[,2:3], method = 'pearson')

x1 x2
x1 1  0 
x2 0  1

`-` $x_1$과 $x_2$의 상관계수가 $0$이므로 상관이 존재하지 않는다

## 8.15

In [102]:
# 8.15
x1 <- c(-5, 4, -1, 2, 2, 3, 3)
x2 <- c(5, 4, 1, -3, -2, -2, -3)
y <- c(11, 11, 8, 2, 5, 5, 4)
m0 <- lm(y~1)

In [103]:
# 1)
m1 <- lm(y~x1+x2)
anova(m0, m1)

Res.Df RSS       Df Sum of Sq F        Pr(>F)     
1 6      73.714286 NA     NA          NA          NA
2 4       3.101281  2 70.613    45.53796 0.001770023

`-` p-value가 매우 작으므로 회귀모형은 유의하다

In [112]:
# 2)
m2 <- lm(y~x1)
m3 <- lm(y~x2)

R_b0 = length(y) * mean(y)^2
R_b0 # R(b0)

[1] 302.2857

$R(\beta_0)=302.285714285714$

In [113]:
anova(m0, m1) # R(b1,b2|b0)
anova(m0, m2) # R(b1|b0)
anova(m0, m3) # R(b2|b0)

Res.Df RSS       Df Sum of Sq F        Pr(>F)     
1 6      73.714286 NA     NA          NA          NA
2 4       3.101281  2 70.613    45.53796 0.001770023

Res.Df RSS      Df Sum of Sq F        Pr(>F)   
1 6      73.71429 NA       NA        NA        NA
2 5      57.83495  1 15.87933  1.372815 0.2941169

Res.Df RSS      Df Sum of Sq F        Pr(>F)      
1 6      73.71429 NA       NA        NA           NA
2 5       3.69958  1 70.01471  94.62521 0.0001951318

$R(\beta_1,\beta_2\mid \beta_0)=70.613\\
R(\beta_1\mid\beta_0)=15.87933\\
R(\beta_2\mid\beta_0)=70.01471$

In [114]:
# 3)
anova(m3, m1)

Res.Df RSS      Df Sum of Sq F         Pr(>F)   
1 5      3.699580 NA       NA         NA        NA
2 4      3.101281  1 0.598299  0.7716799 0.4292997

`-` p-value가 0.429로 0.05보다 작으므로 영가설을 기각하지 못한다

`-` $\beta_1$은 $0$이 아니라고 할 수 없다 

`-` $x_1$은 회귀모형에서 유의하지 않다

In [115]:
# 4)
anova(m2, m1)

Res.Df RSS       Df Sum of Sq F        Pr(>F)     
1 5      57.834951 NA       NA        NA          NA
2 4       3.101281  1 54.73367  70.59492 0.001098159

`-` p-value가 0.001이므로 $\alpha =0.05,0.01$에서 영가설을 기각한다

`-` $\beta_2$는 $0$이 아니다

`-` $x_2$은 회귀모형에서 유의하다

## 8.16

$\boldsymbol{y=X\beta+\epsilon,\;\epsilon}\sim N(0,I),\; \bf{X}$는 $n\times (k+1)$

### 1)

$Q_1=\boldsymbol{y'[I-X(X'X)^{-1}X']y = y'Cy}=SSE$ 이고 $df_E=n-k-1$이고 $\sigma^2=1$이다

$\boldsymbol{V=I}$이고 $\boldsymbol{CVCV=CV}$로 멱등행렬이다

$\begin{aligned}\boldsymbol{(I-X(X'X)^{-1}X')(I-X(X'X)^{-1}X')}&=\boldsymbol{I-2X(X'X)^{-1}X'+X(X'X)^{-1}X'X(X'X)^{-1}X'}\\
&=\boldsymbol{I-X(X'X)^{-1}X'}\end{aligned}$

이므로 $\boldsymbol{I-X(X'X)^{-1}X'}$은 멱등행렬이다

$r(\boldsymbol{C})=tr(\boldsymbol{C})=n-k-1$이므로 $Q_1$은 $\chi^2(n-k-1)$을 따른다

### 2)

$Q_2=\boldsymbol{y'X(X'X)^{-1}X'y = y'By}$ 이고 $df=k+1$이고 $\sigma^2=1$이다

$\boldsymbol{V=I}$이고 $\boldsymbol{BVBV=BV}$로 멱등행렬이다

$\boldsymbol{(X(X'X)^{-1}X')(X(X'X)^{-1}X')= XB^{-1}B(X'X)^{-1}X'=X(X'X)^{-1}X'}$

이므로 $\boldsymbol{X(X'X)^{-1}X'}$은 멱등행렬이다

$r(\boldsymbol{B})=tr(\boldsymbol{B}
)=k+1$이므로 $Q_2$은 $\chi^2(k+1,\frac{1}{2}\boldsymbol{\beta'X'X\beta})$을 따른다

### 3)

$\boldsymbol{BVC}$가 0이면 $Q_1,Q_2$는 서로 독립이다

$\begin{aligned}\boldsymbol{BVC}&=[\boldsymbol{I-X(X'X)^{-1}X']I}\sigma^2\boldsymbol{X(X'X)^{-1}X'}\\
&=\sigma^2[\boldsymbol{I-X(X'X)^{-1}X']}\boldsymbol{X(X'X)^{-1}X'}\\
&=\sigma^2[\boldsymbol{X(X'X)^{-1}X'-(X(X'X)^{-1}X')(X(X'X)^{-1}X')]}\end{aligned}$

그런데 $\boldsymbol{X(X'X)^{-1}X'}$는 멱등행렬이므로 $\boldsymbol{(X(X'X)^{-1}X')(X(X'X)^{-1}X')}=\boldsymbol{X(X'X)^{-1}X'}$

$\sigma^2[\boldsymbol{X(X'X)^{-1}X'-(X(X'X)^{-1}X')(X(X'X)^{-1}X')]}=\sigma^2[\boldsymbol{X(X'X)^{-1}X'}-\boldsymbol{X(X'X)^{-1}X'}]=\boldsymbol{0}$

그러므로 $Q_1,Q_2$는 서로 독립이다

### 4)

$Q_1,Q_2$는 서로 독립이고 $\boldsymbol{\beta}=\boldsymbol{0}$ 이면 $Q_1\sim\chi^2(n-k-1),\;Q_2\sim\chi^2(k+1,\boldsymbol{0})=\chi^2(k+1)$ 

이므로 $\cfrac{Q_2 \big/(k+1)}{Q_1\big/(n-k-1)}\sim F(k+1,n-k-1)$ 이다 